presto-main/src/main/java/io/prestosql/cost/CachingCostProvider.java

```diff
@@ -40,7 +40,7 @@
     private final Session session;
     private final TypeProvider types;
 
-    private final Map<PlanNode, PlanNodeCostEstimate> cache = new IdentityHashMap<>();
```

Comment:
It would be nice if we could add cumulative costs (or whatever we call it) of plan subsets (not necessarily from leaves) to compare alternative plans in optimizer heuristics.

Do you think we could abstract specialized add methods that add two cumulative costs, e.g:
* `addParallelCumulativeCost`
* `addSequentialCumulativeCost`
or similar?